doc/Autotuner.md

```diff
@@ -0,0 +1,306 @@
+# SYCL-BLAS Autotuner Developer Documentation
+This documentation aims to cover the inner workings of the auto tuner from a developer perspective.
+It should be useful if you need to work on or extend the autotuner in future, or just want to better understand how it works.
+
+For information on building and using the autotuner please see the readme [here](../tools/auto_tuner/README.md).
+# Table of Contents
+- [**Overview**](#overview)
+- [**C++ Binaries**](#c++-binaries)
+- [**Parameter Configurations**](#parameter-configurations)
+- [**Python Generation**](#python-generation)
+    - [generated_combinations.def](#generated_combinations.def)
+    - [Instantiation Source Files](#instantiation-source-files)
+- [**CMake**](#cmake)
+- [**Common Tasks**](#common-tasks)
+    - [Adding a new set of configurations](#adding-a-new-set-of-configurations)
+
+# Overview
+The Autotuner is comprised of several parts:
+- The `C++ binaries` that are run by the user to do the tuning. 
+Comprised of 5 different binaries which benchmark different combinations of transposition of the inputs and print the results.
```

Comment:
I see there's a good reason for there being 5 binaries, but I think we still need to be careful since documentation like this can quickly go out of date. Maybe try to unify the way it's referred to so that it's easier to grep for:
```suggestion
Comprised of five different binaries which benchmark different combinations of transposition of the inputs and print the results.
```
However this is a minor concern for this specific PR.